src/recipes/supportedBrowserPage.js

```diff
@@ -0,0 +1,97 @@
+/* 
+    RECIPE: SupportedBrowserPage
+    -------------------------------------------------------------
+    Author: Malick Sere, Lia Hiscock, Joel Ramos, Morgan Graham
+    Description: This recipe looks for strings that would indicate that a page is a "supported browser" page.
+*/
+
+
+
+void function () {
+    window.CSSUsage.StyleWalker.recipesToRun.push(function SupportedBrowserPage(element, results){
+        //doesn't go to microsoft sites
+        if(window.location.href.toString().indexOf("microsoft.com") !== -1 || window.location.href.toString().indexOf("forum") !== -1){
+            return results;
+        }
+        function isVisible(element){
+            //checks if width/height = 0 and left/top < 0
+            if (element.getBoundingClientRect() !== null){
+                var box = element.getBoundingClientRect();
+                var docEl = document.documentElement;
+                var scrollTop = docEl.scrollTop;
+                var scrollLeft = docEl.scrollLeft;
+                var clientTop = docEl.clientTop;
+                var clientLeft = docEl.clientLeft;
+                var width = box.width;
+                var height = box.height;
+                var top = box.top + scrollTop - clientTop;
+                var left = box.left + scrollLeft - clientLeft;
+                var bottom = top + height;
+                var right = left + width;
+                if (width == 0 || height == 0 || bottom <= 0 || right <= 0) {
+                    return 0;
+                }
+            }
+
+            //checks for visibility with computed style
+            var elStyle = getComputedStyle(element);
+            if(elStyle.getPropertyValue("display") === "none"){
+                return 0;
+            }
+            else if(elStyle.getPropertyValue("opacity") < 0.1){
+                return 0;
+            }
+            else if(elStyle.getPropertyValue("transform").includes(" 0,") || elStyle.getPropertyValue("transform").includes(" 0)")){
+                return 0;
+            }
+            else if(elStyle.getPropertyValue("visibility") === "hidden"){
+                return 0;
+            }
+
+            // if text is within an iframe that does not appear: <iframe frameBorder="0" src="">Browser not compatible.</iframe>
+            var elAbove = element;
+            do{
+                if(elAbove.nodeName === "IFRAME"){
+                    if(getComputedStyle(elAbove).getPropertyValue("src") === "" && getComputedStyle(elAbove).getPropertyValue("frameBorder") === 0){
+                        return 0;
+                    }
+                }
+                if(elAbove.parentElement !== null){
+                    elAbove = elAbove.parentElement;
+                }
+
+            }while(elAbove.parentElement !== null);
+            return 1;
+        }
+
+        if(element.nodeName !== "HTML" && element.nodeName !== "SCRIPT" && element.nodeName !== "BODY"){
```

Comment:
"STYLE", "OBJECT", "EMBED", "APPLET"
the last tree also for iframe check

!!element.closest("iframe,object,embed,script,style,meta,link,...")